src/components/TextField/tests/TextField.test.tsx

```diff
@@ -812,4 +812,69 @@ describe('<TextField />', () => {
       expect(textField.find(Connected).prop('right')).toEqual(connectedRight);
     });
   });
+
+  describe('clearButton', () => {
+    it('adds a clear button that clears the input value', () => {
```

Comment:
Considering what you're actually testing here is that `onChange()` is called with an empty value maybe separating this into two tests would make more sense?
1. `renders a clear button when given`
1. `calls onChange() with an empty value when the clear button clicked`